File: backend/internal/domain/entities/graph.go
package entities

import (
	"time"
)

type Graph struct {
	Id        string    `bson:"_id,omitempty"` // ID графа (Mongo автоматически генерирует)
	UserId    string    `bson:"userId"`        // ID пользователя, которому принадлежит граф
	Name      string    `bson:"name"`          // Название графа
	CreatedAt time.Time `bson:"createdAt"`     // Дата создания графа
	Nodes     []Node    `bson:"nodes"`         // Массив узлов графа
	Edges     []Edge    `bson:"edges"`         // Массив рёбер графа

}

File: web-app/src/types/graph.ts
import { Node } from '../types/node'
import { Edge } from './edge';
import { BaseResponse } from './response';

export interface GraphData {
  nodes: Node[];
  edges: Edge[];
}

export type ApiGraphItem = Node | Edge;

export interface GraphResponse {
  Status: string,
  Message: string,
  id: string,
  UserId: string,
  Name: string,
  CreatedAt: string,
  nodes: Node[],
  edges: Edge[]
}

export interface CreateUserGraphResponse extends BaseResponse {
    graphId: string,
}